frontend/internal-packages/agent/src/chat/workflow/nodes/designSchemaNode.ts

```diff
@@ -195,35 +168,37 @@ export async function designSchemaNode(
     )
   }
 
-  // Create prompt variables directly
-  const promptVariables: SchemaAwareChatVariables = {
-    schema_text: schemaText,
-    chat_history: formatMessagesToHistory(state.messages),
-    user_message: userMessage,
-  }
+  // Convert messages to BaseMessage array and add user message
+  const messages = [...state.messages, new HumanMessage(userMessage)]
 
   await logAssistantMessage(
     state,
     repositories,
     'Analyzing table structure and relationships...',
   )
 
-  // Use agent's generate method with prompt variables
-  const response = await agent.generate(promptVariables)
-  const result = await handleSchemaChanges(response, state, repositories)
+  const invokeResult = await invokeDesignAgent({ schemaText }, messages)
```

Comment:
Changed `agent.generate()` to `invokeDesignAgent()` and made it a simple function instead of a class.